tests/functional/i/inherit_non_class.py

```diff
@@ -79,3 +79,24 @@ class NotInheritableSlice(slice): # [inherit-non-class]
 
 class NotInheritableMemoryView(memoryview): # [inherit-non-class]
     pass
+
+
+# Subscription of parent class that implements __class_getitem__
+# and returns cls should be allowed.
+class ParentGood:
+    def __class_getitem__(cls):

```

Comment:
This method should accept one argument in addition to `cls`. Otherwise you will get this:
```python
>>> class ParentGood:
...     def __class_getitem__(cls):
...         return cls
... 
>>> class Child1(ParentGood[int]):
...     pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __class_getitem__() takes 1 positional argument but 2 were given
```

Reply:
Thanks @mitya57! I've updated the test cases.